dask/dataframe/io/sql.py

```diff
@@ -0,0 +1,78 @@
+import numpy as np
+import pandas as pd
+import sqlalchemy as sa
+from sqlalchemy import sql
+
+from dask import delayed
+from dask.dataframe import from_delayed
+
+
+def read_sql_table(table, uri, partitions, index_col, limits=None, columns=None,
+                   **kwargs):
+    """
+    Create dataframe from an SQL table.
+
+    Parameters
+    ----------
+    table : string
+        Table name
+    uri : string
+        Full sqlalchemy URI for the database connection
+    partitions : int or list of values
+        Number of partitions, or list of partition edges (number of partitions will be one less than length of list)
+    index_col : string
+        Column which becomes the index, and defines the partitioning. Should
+        be a indexed column in the SQL server, and numerical.
+        Could be a function to return a value, e.g., ``sqlalchemy.func.row_number``, where the function must
+        be something understood by thr DB backend.
+    limits: 2-tuple or None
+        Manually give upper and lower range of values; if None, first fetches max/min from the DB. Upper limit, if
+         given, isnon-inclusive.
+    columns : list of strings or None
+        Which columns to select; if None, gets all; can include sqlalchemy functions, e.g.,
+        ``sql.func.abs(sql.column('value'))``
+    kwargs : dict
+        Additional parameters to pass to `pd.read_sql()`
+
+    Returns
+    -------
+    dask.dataframe
+    """
+    if index_col is None:
+        raise ValueError("Must specify index column to partition on")
+    engine = sa.create_engine(uri)
+    meta = sa.MetaData()
+    table = sa.Table(table, meta, autoload=True, autoload_with=engine)
+    index = table.columns[index_col] if isinstance(index_col, str) else index_col
+    if isinstance(partitions, int):
+        if limits is None:
+            # calculate max and min for given index
+            q = sql.select([sql.func.max(index), sql.func.min(index)]).select_from(table)
+            minmax = pd.read_sql(q, engine)
+            maxi, mini = minmax.iloc[0]
+            if minmax.dtypes['max_1'].kind == "M":
+                partitions = pd.date_range(start=mini, end=maxi,
+                                           freq='%iS' % ((maxi - mini) / partitions).total_seconds()).tolist()
```

Comment:
Does `np.linspace` not work on this?  That would allow us to pull out a bit of logic outside the nested if tree.

For dtype testing see `np.issubdtype(..., np.datetime64)` or `isinstance(..., pd.Timestamp)`